lottie/src/main/java/com/airbnb/lottie/L.java

```diff
@@ -24,20 +19,7 @@
   private static int traceDepth = 0;
   private static int depthPastMaxDepth = 0;
 
-  public static void debug(String msg) {
-    if (DBG) Log.d(TAG, msg);
-  }
-
-  /**
-   * Warn to logcat. Keeps track of messages so they are only logged once ever.
-   */
-  public static void warn(String msg) {
-    if (loggedMessages.contains(msg)) {
-      return;
-    }
-    Log.w(TAG, msg);
-    loggedMessages.add(msg);
-  }
+  public static LottieLogger logger = new LogcatLogger();
```

Comment:
Actually, this won't work. L is `@RestrictTo(LIBRARY)` so it's not accessible to others to set.

I would do something like:
```
public class LottieLogger {
    public static LottieLogger instance = LottieLogger();

    public static d(...) {
        instance.d(...)
    }
    ...

    public void d(...) {
        ...
    }
}
```

Then include javadocs to indicate that you can extend LottieLogger and replace the implementations with your own.

How does that sound?

Reply:
Yes, we did see "@RestrictTo(LIBRARY)" at first, but seems it's not blocking anything as we're able to use it and set the custom logger from our sample project.

What should we do? Could it be useful that L is accessible for integrators to customize some behaviors of Lottie?

Reply:
@dloic L should not be exposed because it's not clear that it is for Lottie.